src/mlpack/methods/ann/layer/policy.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file policy.hpp
+ * @author Chirag Pabbaraju
+ *
+ * Definition of the Policy class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_POLICY_HPP
+#define MLPACK_METHODS_ANN_LAYER_POLICY_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Policy layer, which implements the policy gradient
+ * algorithm. This layer computes the softmax of an input in forward pass, and gradients wrt softmax layer in the backward pass.
+ * This layer is meant to be used as the last layer in the neural network predicting the policy.
+ * 
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template<
+  typename InputDataType = arma::mat,
+  typename OutputDataType = arma::mat
+>
+class Policy

```

Comment:
My initial thought about this layer is, would this layer be useful to anyone writing any generic neural network?  Or should we instead add this specifically to the RL codebase and namespace only, and then add it as a `CustomLayerTypes`?

I am a little concerned that someone will come across this class, but not understand where or why they might use it.  Ideally, a user should be able to quickly figure out what each class is for and how it is used.  (I realize that is an ideal and we will never truly achieve it, but hey, we should try... :))